File: liqp-core/src/main/java/liqp/exceptions/MissingVariableException.java
package liqp.exceptions;

public class MissingVariableException extends RuntimeException {
  private final String variableName;

  public MissingVariableException(String variableName) {
    super(String.format("Variable '%s' does not exists", variableName));

    this.variableName = variableName;
  }

  public MissingVariableException(String rootName, String variableName) {
    super(String.format("Variable '%s' does not exists (at %s)", rootName, variableName));

    this.variableName = rootName;
  }

  public String getVariableName() {
    return variableName;
  }
}

File: liqp-core/src/main/kotlin/liqp/exceptions/MissingVariableException.kt
package liqp.exceptions

import liqp.context.LLog
import liqp.context.LLogSeverity

class MissingVariableException : RuntimeException {
  val variableName: String

  constructor(variableName: String) : super("Variable '$variableName' does not exist") {
    this.variableName = variableName
  }

  constructor(rootName: String, variableName: String) : super("Variable '$variableName' does not exist (at %$rootName)") {
    this.variableName = rootName
  }
}

class MissingVariable(vararg paths: String) : LLog {
  override val severity = LLogSeverity.WARN
  val variableName: String = paths.joinToString(".")
  override fun toString() = "Variable '$variableName' does not exist"
}
